go/consensus/tendermint/apps/scheduler/scheduler.go

```diff
@@ -367,31 +383,44 @@ func (app *schedulerApplication) electCommittee(
 	// Determine the context, committee size, and pre-filter the node-list
 	// based on eligibility and entity stake.
 	var (
-		err      error
-		nodeList []*node.Node
+		err error
 
 		rngCtx       []byte
 		isSuitableFn func(*api.Context, *node.Node, *registry.Runtime) bool
-
-		workerSize, backupSize, minPoolSize int
 	)
 
+	groupSizes := make(map[scheduler.Role]int)
 	switch kind {
 	case scheduler.KindComputeExecutor:
 		rngCtx = RNGContextExecutor
 		isSuitableFn = app.isSuitableExecutorWorker
-		workerSize = int(rt.Executor.GroupSize)
-		backupSize = int(rt.Executor.GroupBackupSize)
-		minPoolSize = int(rt.Executor.MinPoolSize)
+		groupSizes[scheduler.RoleWorker] = int(rt.Executor.GroupSize)
+		groupSizes[scheduler.RoleBackupWorker] = int(rt.Executor.GroupBackupSize)
```

Comment:
this makes me wonder about moving the GroupSize and GroupBackupSize into the SchedulingConstraints, like this change does with MinPoolSize. are there other places where we like having it separate like this?

Reply:
He he, I had the exact same thought as well. It would definitely make things nicer.